test/kubernetes/e2e/features/http_tunnel/suite.go

```diff
@@ -0,0 +1,153 @@
+package http_tunnel
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/solo-io/gloo/pkg/utils/kubeutils"
+	"github.com/solo-io/gloo/pkg/utils/requestutils/curl"
+	"github.com/solo-io/gloo/projects/gateway/pkg/defaults"
+	"github.com/solo-io/gloo/test/gomega/matchers"
+	"github.com/solo-io/gloo/test/kubernetes/e2e"
+	testDefaults "github.com/solo-io/gloo/test/kubernetes/e2e/defaults"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	_ "embed"
+)
+
+const (
+	httpbinExampleCom = "httpbin.example.com"
+)
+
+var _ e2e.NewSuiteFunc = NewTestingSuite
+
+//go:embed testdata/squid.yaml
+var squidYaml []byte
+
+//go:embed testdata/proxy.yaml
+var proxyYaml []byte
+
+//go:embed testdata/edge.yaml
+var edgeYaml []byte
+
+//go:embed testdata/gateway.yaml
+var gatewayYaml []byte
+
+// testingSuite is the entire Suite of tests for the HTTP Tunnel feature
+type testingSuite struct {
+	suite.Suite
+	ctx              context.Context
+	testInstallation *e2e.TestInstallation
+}
+
+func NewTestingSuite(
+	ctx context.Context,
+	testInst *e2e.TestInstallation,
+) suite.TestingSuite {
+	return &testingSuite{
+		ctx:              ctx,
+		testInstallation: testInst,
+	}
+}
+
+func (s *testingSuite) SetupSuite() {
+	err := s.testInstallation.Actions.Kubectl().Apply(s.ctx, squidYaml)
```

Comment:
Could using `ApplyFile` work instead ? That is the usual convention used in our tests

Reply:
To avoid boilderplate, the [BaseTestingSuite](https://github.com/solo-io/gloo/blob/main/test/kubernetes/e2e/tests/base/base_suite.go) can be used - [example](https://github.com/solo-io/gloo/blob/b233d00d313b6d084fc1500d18db79425e346e1d/test/kubernetes/e2e/features/services/httproute/suite.go#L23)

Reply:
I'm trying to avoid what I see in the `types.go` files, which is common with `BaseTestingSuite`. I think it's desirable it be easy to see what is being applied in the before/after suite/test and not have to consult a map in another file.

Reply:
I often see tests that apply some things using a map in `types.go` and then apply more stuff in the test suite methods making it harder to reason about what is happening.

Reply:
Would a followup then include moving all test over to using embed? I'm trying to not go off the precedent that's already in place

Reply:
That is a lot of changes and considering we are talking about tests, I don't view moving everything to one pattern or another as something that has to be done all at once; the savings just isn't there. If a way we were doing something was creating a consistent drain on our time, them I would be all for shaving the yak more aggressively.

There is a [linting ticket that includes introducing a technique](https://github.com/solo-io/solo-projects/issues/7277#issuecomment-2641342652) that will allow us to shave these yaks overtime without allowing further introduction of deprecated ways of doing things. Concretely, there are linters that that can added to fail we are touching files in specific (sub)directories that use `TestInstallation.Assertion` or `util.MustGetThisDir`.